mirgecom/steppers.py

```diff
@@ -214,13 +214,11 @@ def _advance_state_leap(rhs, timestepper, state,
         if isinstance(state, np.ndarray):
             state = thaw(freeze(state, actx), actx)
 
-        if dt < 0:
-            return istep, t, state
-
         if pre_step_callback is not None:
             state, dt = pre_step_callback(state=state,
                                           step=istep,
                                           t=t, dt=dt)
+        stepper_cls.dt = dt
```

Comment:
I think this can/should be moved into the if-block directly above it. The timestep the stepper class has only needs to be updated if dt is changed.

Reply:
(68e1e60)